scalafmt-core/shared/src/main/scala/org/scalafmt/config/ProjectFiles.scala

```diff
@@ -65,7 +64,7 @@ object ProjectFiles {
     }
 
     private def create(seq: Seq[String], f: String => PathMatcher) = seq
-      .map(_.asFilename).distinct.map(f)
```

Comment:
could you explain this part, please? was this change not working otherwise?

will PathMatcher work correctly on forward slashes? the user is allowed to specify forward slashes, but the PathMatcher must be able to match the real file names, which on Windows would contain backslashes.

Reply:
Yes, having forward matches in a path matcher will match both files with backslashes and forward slashes on Windows (both on Scala Native and JVM, just tested this on my machine). The problem I was having is the fact that on Scala Native on Windows PathMatchers with backslashes would not match anything (likely due to a bug there), and this is the simplest fix (of course, I could separately move this behavior to a compat object, like I did with other things, but since this works both on JVM and Native I made the change for both).

Reply:
hmm... that issue was due to one backslash instead of two. don't know.

but, at least for native, perhaps what you need to do is actually do the opposite: replace all backslashes with slashes for the path matcher.

Reply:
Yes, I brought back the asFilename calls, just changed its behavior on native to do the opposite. It's not very useful for config files, since sconfig can completely fail to parse when the content contains backslashes (so before we reach the .asFilename calls), but perhaps it's also used elsewhere (so the change might still be useful)